library/board.py

```diff
@@ -21,11 +21,24 @@ def __init__(self) -> None:
 
     def check_board_victory(self) -> typing.Optional[str]:
         """Determine if the entire board has been won, and by who."""
-        grid_states = np.array(
-            (self.check_subboard_victory(i) for i in "789456123"), dtype=np.str_
-        )
-        # This might have off ordering, TODO on testing it
-        grid_states.reshape((3, 3))
+        grid_states = np.full((3, 3), "·")
+        grid_map = {
+            1: [2, 0],
+            2: [2, 1],
+            3: [2, 2],
+            4: [1, 0],
+            5: [1, 1],
+            6: [1, 2],
+            7: [0, 0],
+            8: [0, 1],
+            9: [0, 2],
+        }
+        for i in range(1, 10):
+            working_space_location = grid_map[i]
```

Comment:
you could maybe do `y, x = grid_map[i]` to make it clearer how we are indexing (then do `grid_states[y, x]` on lines 38-40)

that might (?) help code readability.

Reply:
Changed this code to this:
```
        for i in range(1, 10):
            y, x = grid_map[i]
            grid_states[y, x] = self.check_subboard_victory(str(i))
```